### PR TITLE
DataViews: add labels to "in-filters"

### LINKED ISSUE #55963: Add a new `LabelPosition` of `inside` to `InputControl`
## What problem does this address?

A use-case has emerged for [`InputControl`](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/input-control) to put the label inside of the control wrapper.

A label and input, all contained within a single border

This is not currently possible to do properly, and as such the consumer is using the `prefix` prop instead. This means that the input does not have an accessible name, and that part of the field is not clickable.

## What is your proposed solution?

A new [`LabelPosition`](https://github.com/WordPress/gutenberg/blob/e4c8ef8456bfe4c15ebd505dcb2ccf572eeab071/packages/components/src/input-control/types.ts#L20) of `inside` should be added to `InputControl`, to allow consumers to put the label inside of the field, and not misuse the `prefix` option. This will better ensure that inputs are appropriately named.

### PR DESCRIPTION
**Pull request by andrewhayward**
## What?
This patch ensures that "in-filter" inputs have accessible names.

It is an alternative approach to that explored by #55977, tackling the issue closer to the problem. Given the likely short-term lifecycle of the filter implementation, this is more economical alternative to #55963.

## Why?
All user inputs should have an [accessible name](https://developer.mozilla.org/en-US/docs/Glossary/Accessible_name). This is currently lacking in data view filters.

## How?
The visible label is updated to use a `<label>` element, and semantically linked to the select input with a `for`/`id` combination.

The alternative of setting `label`/`hideLabelFromVision` props on the parent `<SelectControl>` was considered, but this approach has the benefit of physically associating the label with the control as well as semantically, so clicking on it will focus the input.

## Testing Instructions
Nothing should visually change.

Using developer tools, a screen reader, or similar, confirm that the filter inputs have accessible names, derived from the visible labels.

Data views filters, with developer tools information below illustrating the relationship between a label and an input, circled in red.

### CODE REVIEW

File: packages/edit-site/src/components/dataviews/in-filter.js
Comment (youknowriad) [inline packages/edit-site/src/components/dataviews/in-filter.js:33]: Aside: Just saw this, I think we should either remove the `:` or use sprint and __ here, not all locals support this notation I think.  cc @oandregal

File: packages/edit-site/src/components/dataviews/in-filter.js
Comment (oandregal) [inline packages/edit-site/src/components/dataviews/in-filter.js]: Nice, thanks. Could we use as `id` something like `dataviews__filters-in-author` instead of `dataviews__fisters-in-0`?

We could take the field name from `filter.field` (`author`, `status`) and they are unique. `filter.name` is more of a user-facing name (can have spaces, etc.) so it wouldn't work that well I guess.

File: packages/edit-site/src/components/dataviews/in-filter.js
Comment (oandregal) [reply packages/edit-site/src/components/dataviews/in-filter.js:33]: 👍 This is going to become "Author is" when the new component is ready.

File: packages/edit-site/src/components/dataviews/in-filter.js
Comment (youknowriad) [reply packages/edit-site/src/components/dataviews/in-filter.js]: Should we use React's useId instead of useInstanceId? Feels like the latter should be deprecate these days.

**Review by oandregal — APPROVED** [review]
I left a suggestion for the `id`, though any works for me – whatever you think makes more sense.

File: packages/edit-site/src/components/dataviews/in-filter.js
Comment (oandregal) [reply packages/edit-site/src/components/dataviews/in-filter.js:33]: I'll wait until this PR is merged to fix it – or if you'd rather do it yourself, @andrewhayward, you're welcome :)

File: packages/edit-site/src/components/dataviews/in-filter.js
Comment (andrewhayward) [reply packages/edit-site/src/components/dataviews/in-filter.js]: Happy to go with whatever works for the team. If we're confident that `filter.field` is unique, and each filter type will only appear once on the page, then makes sense to use that – in which case, we won't need to apply either `useInstanceId` or `useId`.

File: packages/edit-site/src/components/dataviews/in-filter.js
Comment (andrewhayward) [reply packages/edit-site/src/components/dataviews/in-filter.js:33]: @oandregal I'll let you do that in a separate PR. Scope, etc!

Comment (oandregal) [reply packages/edit-site/src/components/dataviews/in-filter.js:33]: https://github.com/WordPress/gutenberg/pull/56075

**Pull request merged**
